devops/tests/test_component.py

```diff
@@ -24,11 +24,35 @@
           image: imagined.registry.tld/myproj-service-test-deployment:latest
 """
 
+CRONJOB = """
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: test-cronjob
+spec:
+  schedule: '* * * * *'
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            app: test-cronjob
```

Comment:
@victorv-sbp: I checked the kubernetes documentation for the CronJob kind, and found out that there is no `metadata` under the `template:` key in their example (https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#creating-a-cron-job). Is there a reason why this is defined here, or is it just by accident?

Reply:
Indeed, it does not show in any of the official examples (probably in order to only display and underline the critical fields for a cronjob), but it is a valid yaml definition of a CronJob(kubeval, PyCharm k8s yaml validator and applying the definition on a cluster, all work with a `metadata` under the `template`). 
Also, if you apply a CronJob on a cluster without that metadata branch, you can later see that describing that CronJob will display a `metadata` field under the `template` area.

In conclusion the `metadata` area is not wrong but it's also not really important for a Job definition. It could be deleted without having any effect on the test, but considering that the deployment also has a metadata (with the same role and importance), I would keep it there just for consistency. If you still want to delete it then I will do so.

Reply:
I'd rather not have anything excessive if it's not needed nor used anyway. But I'm OK to have these in the tests, so that it's certain that the tests pass.